pkg/dao/schema/workflow.go

```diff
@@ -0,0 +1,94 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/seal-io/walrus/pkg/dao/entx"
+	"github.com/seal-io/walrus/pkg/dao/schema/intercept"
+	"github.com/seal-io/walrus/pkg/dao/schema/mixin"
+	"github.com/seal-io/walrus/pkg/dao/types/object"
+)
+
+type Workflow struct {
+	ent.Schema
+}
+
+func (Workflow) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		mixin.Metadata(),
+	}
+}
+
+func (Workflow) Indexes() []ent.Index {
+	return []ent.Index{
+		index.Fields("project_id", "name").
+			Unique(),
+	}
+}
+
+func (Workflow) Fields() []ent.Field {
+	return []ent.Field{
+		object.IDField("project_id").
+			Comment("ID of the project that this workflow belongs to.").
+			NotEmpty().
+			Immutable(),
+		object.IDField("environment_id").
+			Comment("ID of the environment that this workflow belongs to.").
+			Optional().
+			Immutable(),
+		field.String("type").
+			Comment("Type of the workflow.").
+			NotEmpty().
+			Immutable(),
+		field.Int("parallelism").
+			Comment("Number of task pods that can be executed in parallel of workflow.").
+			Positive().
+			Default(10),
+		field.Int("timeout").
+			Comment("Timeout seconds of the workflow.").
+			NonNegative().
+			Default(0),
+		field.Int("version").
+			Comment("Execution version of the workflow.").
+			NonNegative().
+			Default(0).
```

Comment:
what does the default version 0 mean? `WorkflowExecution` doesn't have a default config?

Reply:
Version means the current execution sequence. When a workflow is created, there is no workflow execution. Therefore the version of the workflow is zero. When workflow execution created, the version will add 1.